src/google/protobuf/compiler/cpp/file.cc

```diff
@@ -556,7 +556,7 @@ void FileGenerator::GenerateSourceDefaultInstance(int idx, io::Printer* p) {
             };
           };
 
-          PROTOBUF_ATTRIBUTE_NO_DESTROY PROTOBUF_CONSTINIT
+          PROTOBUF_ATTRIBUTE_NO_DESTROY PROTOBUF_CONSTINIT$ dllexport_decl$
```

Comment:
shouldn't the space be before the `$`?  I'm a little surprised this works but it seems to...

Reply:
` $foo$` means "add a space and expand foo"
`$ foo$` means "add a leading space if the expansion of foo is non-empty and expand foo"
`$foo $` means "add a trailing space if the expansion of foo is non-empty and expand foo"